src/auth/auth.service.ts

```diff
@@ -40,6 +33,22 @@ export class AuthService {
     return userRecordWithoutPassword;
   }
 
+  private signUser(id: string, role: RoleType) {
+    return this.jwtService.sign({
+      sub: id,
+      claim: role,
+    });
+  }
+
+  public googleLogin(req: any) {
+    if (!req.user) return "No user from google";
+
+    //TODO?: Save user in database
```

Comment:
Yep we need to save the user into the DB. To do this we need to make some changes to the `UserEntity` model, to add probably 2 more fields, those being `provider` and `providerId`. Then we can save the user and return the `userRecord` from DB along with `JWT` token

PS: I have coded this function on my local when I pulled down your changes so if you want we can have a talk later on VC.

Reply:
Got it, I saw the previous message. Later on we could chat if you are available